alerting/src/main/kotlin/org/opensearch/alerting/InputService.kt

```diff
@@ -107,6 +125,35 @@ class InputService(
         }
     }
 
+    /**
+     * Extends the given query builder with query that filters the given indices with the given doc ids per index
+     * Used whenever we want to select the documents that were found in chained delegate execution of the current workflow run
+     *
+     * @param query Original bucket monitor query
+     * @param indexToDocIds Map of finding doc ids grouped by index
+     */
+    private fun updateInputQueryWithFindingDocIds(
+        query: QueryBuilder,
+        indexToDocIds: Map<String, List<String>>,
+    ): QueryBuilder {
+        val queryBuilder = QueryBuilders.boolQuery().must(query)
+        val shouldQuery = QueryBuilders.boolQuery()
+
+        indexToDocIds.forEach { entry ->
+            shouldQuery
+                .should()
+                .add(
+                    BoolQueryBuilder()
+                        .must(MatchQueryBuilder("_index", entry.key))
+                        .must(TermsQueryBuilder("_id", entry.value))
```

Comment:
the max findings would be 10000.

Reply:
we know definitively that this would be the size of the terms

Reply:
@lezzago - not sure that it will explode and I think that we are already doing that -> indexToDocIds is a map where the key is the index we are searching while the value is the list of docIds. Maybe I should just renamed the parameter to something more obvious like:
documentIdsPerIndex